src/main/scala/ch/epfl/bluebrain/nexus/admin/projects/Projects.scala

```diff
@@ -0,0 +1,149 @@
+import ch.epfl.bluebrain.nexus.commons.types.identity.Identity
+import ch.epfl.bluebrain.nexus.sourcing.Aggregate
+import ch.epfl.bluebrain.nexus.sourcing.akka._
+
+class Projects[F[_]](agg: Agg[F], index: Index)(implicit F: Monad[F], http: HttpConfig, clock: Clock) {
+
+  /**
+    * Creates a project.
+    *
+    * @param label       the project label (segment)
+    * @param description an optional description
+    * @param caller      an implicitly available caller
+    * @return project the created project resource metadata if the operation was successful, a rejection otherwise
+    */
+  def create(label: ProjectLabel, description: Option[String])(implicit caller: Identity): F[ProjectMetaOrRejection] =
+    index.getOrganization(label.organization) match {
+      case Some(org) =>
+        index.getProject(label) match {
+          case None =>
+            val projectId = UUID.randomUUID
+            val command   = CreateProject(projectId, org.id, label, description, Meta(caller, clock.instant))
+            eval(projectId.toString, command)
+          case Some(_) => F.pure(Left(ProjectAlreadyExists))
+        }
+      case None => F.pure(Left(OrganizationDoesNotExist))
+    }
+
+  /**
+    * Updates a project.
+    *
+    * @param label       the project label (segment)
+    * @param description an optional description
+    * @param caller      an implicitly available caller
+    * @return the updated project resource metadata if the operation was successful, a rejection otherwise
+    */
+  def update(label: ProjectLabel, description: Option[String])(implicit caller: Identity): F[ProjectMetaOrRejection] =
+    index.getProject(label) match {
+      case Some(project) =>
+        if (project.deprecated) F.pure(Left(ProjectIsDeprecated))
+        else
+          eval(project.id.toString,
+               UpdateProject(project.id, project.label, description, project.rev, Meta(caller, clock.instant)))
+      case None => F.pure(Left(ProjectDoesNotExists))
+    }
+
+  /**
+    * Deprecates a project.
+    *
+    * @param label   the project label (segment)
+    * @param rev     the project revision
+    * @param caller  an implicitly available caller
+    * @return the deprecated project resource metadata if the operation was successful, a rejection otherwise
+    */
+  def deprecate(label: ProjectLabel, rev: Long)(implicit caller: Identity): F[ProjectMetaOrRejection] =
+    index.getProject(label) match {
+      case Some(project) =>
+        if (project.deprecated) F.pure(Left(ProjectIsDeprecated))
+        else eval(project.id.toString, DeprecateProject(project.id, rev, Meta(caller, clock.instant)))
+      case None => F.pure(Left(ProjectDoesNotExists))
+    }
+
+  /**
+    * Fetches a project from the aggregate.
+    *
+    * @param id the project permanent identifier
+    * @return Some(project) if found, None otherwise
+    */
+  def fetch(id: UUID): F[ProjectResourceOrRejection] = agg.currentState(id.toString).map {
```

Comment:
if we deprecate and update by using project label, shouldn't we also be able to fetch project by label rather than id?